source/tls_channel_handler.c

```diff
@@ -326,54 +299,87 @@ void aws_tls_ctx_options_set_minimum_tls_version(
     options->minimum_tls_version = minimum_tls_version;
 }
 
-/**
- * The following API is deprecated due to inconsistent override versus append system/library trust
- * behavior between Windows, Mac, and Unix based platforms
- */
 int aws_tls_ctx_options_override_default_trust_store_from_path(
     struct aws_tls_ctx_options *options,
     const char *ca_path,
     const char *ca_file) {
 
+    struct aws_string *ca_path_tmp = NULL;
+    struct aws_byte_buf ca_file_tmp;
```

Comment:
Let's add a note here that on success, ca_file_tmp gets copied into a persistent structure by value, so there's no need to clean it up.  This is sort of a weird (in that we don't do it often) pattern but I can see its value in making the change atomically all-or-nothing.